crates/goose/src/providers/types/content.rs

```diff
@@ -15,78 +17,52 @@ impl Text {
     }
 }
 
-// Tool use content
+
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ToolUse {
+/// A request for a tool call from the agent
+pub struct ToolRequest {
+    /// A unique identifier for this tool request
     pub id: String,
-    pub name: String,
-    pub parameters: Value,
-    #[serde(default)]
-    pub is_error: bool,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub error_message: Option<String>,
+    /// The requested tool call, which is a Result because it may be malformed
+    pub call: AgentResult<ToolCall>,
```

Comment:
i found this a bit confusing because the AgentError here can only be either ToolNotFound or InvalidParameters. Don't know if its breaking that down:
```
AgentError
   -> ToolRequestError
         - ToolNotFound
         - InvalidParameters
   -> ToolResponseError
         - InvalidParameters
         - ExecutionError
    - InternalError
```

This would make handling error cases easier if we wanna handle them upstream for retries

Reply:
This does make sense, im going to explore as we implement the agent loop here. So will merge for now but revisit